### PR TITLE
Use math function in Kokkos namespace for raytracing example if available

### PR DESCRIPTION
**Pull request by masterleinad**
With `Kokkos` release 3.7, math functions are declared in the `Kokkos` namespace with aliases in the `Kokkos::Experimental` namespace only if `Kokkos` was configured with `Kokkos_ENABLE_DEPRECATED_CODE_3=ON`.

### CODE REVIEW

**Review by aprokop — COMMENTED** [review]
I think `ArborX_DetailsKokkosExtMathFunctions.hpp` would the the right place to put those things in, instead of the driver.

**Comment by masterleinad** [comment]
> I think `ArborX_DetailsKokkosExtMathFunctions.hpp` would the the right place to put those things in, instead of the driver.

Here you go. I don't necessarily agree that that's the better way to do it, though.

File: src/kokkos_ext/ArborX_DetailsKokkosExtMathFunctions.hpp [hunk 1]
```diff
@@ -17,10 +17,20 @@
 namespace KokkosExt
 {
 
-#if KOKKOS_VERSION >= 30699
+#if KOKKOS_VERSION >= 30700
```
Comment (dalg24) [inline src/kokkos_ext/ArborX_DetailsKokkosExtMathFunctions.hpp:20]: Why did you change the guards?

Comment (aprokop) [reply src/kokkos_ext/ArborX_DetailsKokkosExtMathFunctions.hpp:20]: I actually like this change. It will work with our existing testing, and I think it's better in the medium term.

Comment (dalg24) [reply src/kokkos_ext/ArborX_DetailsKokkosExtMathFunctions.hpp:20]: > It will work with our existing testing

How does `30699` not work?

>  I think it's better in the medium term.

What is better about it?

**Comment by aprokop** [comment]
So it errors out with
```
no member named 'fabs' in namespace 'Kokkos'; did you mean 'abs'? [clang-diagnostic-error]
using Kokkos::fabs;
```
This is the same issue I had [here](https://github.com/arborx/ArborX/pull/713#discussion_r949428612). So I don't understand whether it's correct to say `With Kokkos release 3.7, math functions are declared in the Kokkos namespace` or whether it is missing an include.

File: src/kokkos_ext/ArborX_DetailsKokkosExtMathFunctions.hpp [hunk 2]
```diff
@@ -17,10 +17,20 @@
...
+using Kokkos::acos;
+using Kokkos::cos;
+using Kokkos::expm1;
+using Kokkos::fabs;
 using Kokkos::isfinite;
+using Kokkos::sin;
 #else
+using Kokkos::fabs;
```
Comment (dalg24) [inline src/kokkos_ext/ArborX_DetailsKokkosExtMathFunctions.hpp:28]:
```suggestion
using Kokkos::Experimental::fabs;
```

File: src/kokkos_ext/ArborX_DetailsKokkosExtMathFunctions.hpp (hunk 1 above)
Comment (aprokop) [reply src/kokkos_ext/ArborX_DetailsKokkosExtMathFunctions.hpp:20]: > How does `30699` not work?

That's not what I meant. I mean that 30700 would work, too.

> What is better about it?

I like hiding functionality behind the exact releases, rather than some development versions.

File: src/kokkos_ext/ArborX_DetailsKokkosExtMathFunctions.hpp (hunk 2 above)
Comment (masterleinad) [reply src/kokkos_ext/ArborX_DetailsKokkosExtMathFunctions.hpp]: Fixed.

**Comment by dalg24** [comment]
> > I think `ArborX_DetailsKokkosExtMathFunctions.hpp` would the the right place to put those things in, instead of the driver.
> 
> Here you go. I don't necessarily agree that that's the better way to do it, though.

Are you saying that because it is in an example and not in ArborX implementation?  (I overlooked this)

File: src/kokkos_ext/ArborX_DetailsKokkosExtMathFunctions.hpp (hunk 1 above)
Comment (masterleinad) [reply src/kokkos_ext/ArborX_DetailsKokkosExtMathFunctions.hpp:20]: Any commit after creating the release branch for 3.6 (and thus bumping the version on `develop`) would report `30699` but that doesn't mean that the changes necessary are already part of that commit.

Comment (dalg24) [reply src/kokkos_ext/ArborX_DetailsKokkosExtMathFunctions.hpp:20]: The promotion to the main `Kokkos::` namespace is the 1st thing that happened in the 3.7b release cycle so it is safe to do so

**Comment by masterleinad** [comment]
> Are you saying that because it is in an example and not in ArborX implementation? (I overlooked this)

Yes, I think it's better to use documented `Kokkos` features rather than some undocumented wrapper even if that requires checking for the `Kokkos` version. I changed it because that appeared what @aprokop and you wanted (and didn't feel like worth fighting about too much).

File: src/kokkos_ext/ArborX_DetailsKokkosExtMathFunctions.hpp (hunk 1 above)
Comment (masterleinad) [reply src/kokkos_ext/ArborX_DetailsKokkosExtMathFunctions.hpp:20]: That requires special knowledge about the order of commits in the release branch. :slightly_smiling_face:

**Comment by dalg24** [comment]
> > Are you saying that because it is in an example and not in ArborX implementation? (I overlooked this)
> 
> Yes, I think it's better to use documented `Kokkos` features rather than some undocumented wrapper even if that requires checking for the `Kokkos` version. I changed it because that appeared what @aprokop and you wanted (and didn't feel like worth fighting about too much).

I agree with you for an example.  We do not want to showcase use implementation details of ArborX.

> I changed it because that appeared what @aprokop and you wanted (and didn't feel like worth fighting about too much). 

You simply said you did not agree and did not say why.

**Comment by aprokop** [comment]
retest this please
